fabric-data-attachment-api-v1/src/main/java/net/fabricmc/fabric/impl/attachment/AttachmentTargetImpl.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.attachment;
+
+import net.minecraft.nbt.NbtCompound;
+
+import net.fabricmc.fabric.api.attachment.v1.AttachmentTarget;
+
+public interface AttachmentTargetImpl extends AttachmentTarget {
+	/**
+	 * Writes all the attached data to NBT.
+	 *
+	 * @param nbt the NBT to write to
+	 */
+	default void writeAttachmentsToNbt(NbtCompound nbt) {
+		throw new UnsupportedOperationException("Implemented via mixin");
+	}
+
+	/**
+	 * Reads all the attached data from NBT.
+	 *
+	 * @param nbt the NBT to read from
+	 */
+	default void readAttachmentsFromNbt(NbtCompound nbt) {
+		throw new UnsupportedOperationException("Implemented via mixin");
+	}
+
+	/**
+	 * @return whether the {@link AttachmentTarget} has any attachment data to serialize
+	 */
+	default boolean hasPersistentAttachments() {
+		throw new UnsupportedOperationException("Implemented via mixin");
+	}
```

Comment:
As theser are impl methods I would prefix them with `fabric_` to prevent possible conflicts with other mods. (Its unlikely but trivial to do)